src/components/EntityEditor/Code/CodeInputCard.vue

```diff
@@ -137,58 +128,77 @@ const props = defineProps({
 const emit = defineEmits(['update:modelValue', 'update:expanded'])
 
 // eslint-disable-next-line @typescript-eslint/unbound-method
-const { t, te } = useI18n()
+const { t } = useI18n()
 const $q = useQuasar()
 const { notify } = useNotify()
 const showManualForm = ref(false)
 const showHelp = ref(false)
+const codeApi = inject(CodeApiKey)
 
 const disabledCodes = computed(() => {
-  // It is allowed to have the same codes twice, one with scope 'self' and one with 'leaves'.
-  // For simplicity, we just check if a code is already duplicated.
   return props.modelValue.filter((c1, i1) => props.modelValue.some((c2, i2) => codeEquals(c1, c2) && i1 != i2))
 })
 
-function codeEquals(code1?: Code, code2?: Code, includeScope = false) {
+function codeEquals(code1?: Code, code2?: Code) {
   if (!code1 || !code2) return false
   return (
     code1.codeSystem.uri == code2.codeSystem.uri &&
-    code1.code == code2.code &&
-    (!includeScope || code1.scope == code2.scope)
+    code1.code == code2.code
   )
 }
 
-function addEntries(entries?: Code[]) {
-  if (!entries) return [false]
+function addEntries(entries?: CodeWithScope[]) {
+  if (!entries) return Promise.resolve([false])
+  if (!codeApi) return Promise.reject({ message: 'Could not load data from the server.' })
+
   const newModelValue = props.modelValue.slice()
-  const results = entries.map((e) => {
-    if (newModelValue.some((c) => codeEquals(c, e, true))) return false
-    newModelValue.push(e)
+  return Promise.all(entries.map(async({code: code, scope: scope}) => {
+    if (newModelValue.some((c) => codeEquals(c, code))) return false
+    if(scope == CodeScope.Self) {
+      newModelValue.push(code)
+    } else {
+      await codeApi?.getCode(encodeURIComponent(code.uri!), code?.codeSystem?.externalId || '', scope)
+      .then((r) => {
+        newModelValue.push(r.data);
+      })
```

Comment:
I guess, the file should be reformatted.

```suggestion
      .then((r) => newModelValue.push(r.data))
```

Reply:
Addressed in b5e55fe6529f22483decddfca4e168e2f0f9a641 and 2f857c476b76e5488495322eeedbe985e66cefd0